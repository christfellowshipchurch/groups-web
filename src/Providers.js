import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
