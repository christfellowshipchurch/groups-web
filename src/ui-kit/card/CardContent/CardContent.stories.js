import React from 'react';
import { storiesOf } from '@storybook/react';

import CardContent from '.';

storiesOf('ui-kit/card/CardContent', module)
  .addDecorator((story) => (
    <div style={{ width: '50%', background: 'salmon' }}>{story()}</div>
  ))
  .add('default', () => <CardContent>Padded card content</CardContent>);
