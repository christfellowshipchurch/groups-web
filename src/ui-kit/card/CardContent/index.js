import styled from 'styled-components';

import PaddedDiv from '../../PaddedDiv';

const CardContent = styled(PaddedDiv).attrs({
  className: 'ui-kit.card.CardContent',
})`
  padding: ${({ theme }) =>
    `${theme.sizing.baseUnit(1.5)} ${theme.sizing.baseUnit(2)}`};
`;

export default CardContent;
