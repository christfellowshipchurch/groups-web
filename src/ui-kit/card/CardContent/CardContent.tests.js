import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import CardContent from '.';

describe('the CardContent component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <CardContent>Boom!</CardContent>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
