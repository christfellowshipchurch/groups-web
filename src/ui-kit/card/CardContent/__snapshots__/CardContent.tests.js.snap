// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CardContent component should render 1`] = `
.c0 {
  padding: 1rem 1rem;
  padding: 1.5rem 2rem;
}

<div
  className="sc-9taxng-0 c0 ui-kit.PaddedDiv ui-kit.card.CardContent"
>
  Boom!
</div>
`;
